ondewo/nlu/rag.proto

```diff
@@ -0,0 +1,3138 @@
+// Copyright 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Modifications Copyright 2020-2023 ONDEWO GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// File-level comment for <code>ondewo/nlu/rags.proto</code>.
+// This file contains a single service <a href="#ondewo.nlu.Rags">Rags</a>. The Rags service provides integration with RAGFlow for Retrieval-Augmented Generation (RAG), including dataset management, document processing, chunk retrieval, conversational AI with chat and agent assistants, and file management. Key message types include <a href="#ondewo.nlu.Dataset">Dataset</a>, <a href="#ondewo.nlu.Chat">Chat</a>, and <a href="#ondewo.nlu.Agent">Agent</a>.
+syntax = "proto3";
+
+package ondewo.nlu;
+
+import "google/protobuf/empty.proto";
+import "google/protobuf/struct.proto";
+
+// TODO change Github link if changes to RAGFlow are merged to upstream
+
+// Provides an interface for interacting with the RAGFlow RAG engine.
+// Covers all entpoints of the <a href="https://github.com/ondewo/ragflow">RAGFlow</a> HTTP API. For more information on RAGFlow refer to the <a href="https://ragflow.io/docs/dev/">official documentation</a>
+service Rags {
+
+    // ===========================================
+    // Dataset Management (6 methods)
+    // REST: /api/v1/datasets
+    // ===========================================
+
+    // RAGFlow endpoint: POST /api/v1/datasets
+
+    // Create a new dataset (knowledge base).<br>
+    // Uses tenant's default embedding model if not specified.
+    rpc CreateDataset(CreateDatasetRequest) returns (Dataset);
```

Comment:
Some endpoint and message names overlaps existing ones in ondewo.nlu. Technically this is not an issue however from user point of view it is easier to have different names for all endpoints and protobuf messages. 

Can you please prefix every endpoint and protobuf message for the `Rags` service with `Rag`? 

For example: 

From: `rpc CreateDataset(CreateDatasetRequest) returns (Dataset);`
To: ` rpc RagCreateDataset(RagCreateDatasetRequest) returns (RagDataset);`

If there is Rag already in the name of the endpoint or protobuf message, then simplify it in the following way:

From: `rpc ListAgents(ListRagAgentsRequest) returns (RagAgentList);`
To: `rpc RagListAgents(RagListAgentsRequest) returns (RagAgentList);`